src/figuresClasses.ts

```diff
@@ -1,19 +1,66 @@
-export interface Figure {
+function round(num: number): number {
+  return Math.floor(num * 100) / 100;
+}
 
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape;
 
+  constructor(public color: Color, public a: number, public b: number,
+    public c: number) {
+    if (Math.min(a, b, c) <= 0
+    || Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error('Incorrect sides');
+    }
+    this.shape = 'triangle';
+  }
+
+  getArea(): number {
+    const p: number = 0.5 * (this.a + this.b + this.c);
+
+    return round(Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c)));
+  }
 }
 
 export class Circle implements Figure {
+  shape: Shape;
 
+  constructor(public color: Color, public radius: number) {
+    if (Math.min(radius) <= 0) {
+      throw new Error('Incorrect radius');
```

Comment:
Please be more descriptive in error messages. Please add f.e. 'Should be greater than 0' to the current message